chrome/content/zotero/xpcom/utilities.js

```diff
@@ -1527,14 +1484,14 @@ Zotero.Utilities = {
 	 */
 	"itemToCSLJSON":function(zoteroItem) {
 		if (zoteroItem instanceof Zotero.Item) {
-			zoteroItem = zoteroItem.toArray();
+			zoteroItem = Zotero.Utilities.Internal.itemToExportFormat(zoteroItem);
```

Comment:
I don't think `itemToCSLJSON` is meant to handle notes and attachments (though I could very well be wrong. Can citeproc/CSL handle notes or attachments in any reasonable way), primarily because [`CSL_TYPE_MAPPINGS`](https://github.com/zotero/zotero/pull/659/files#diff-a098326f7cdaa37adf1398d92b384be5R103) does not map these "item" types. We do fall back to "article" in those cases, but I don't think that is the right thing to do. For unrecognized types, we should be throwing an exception here.

Attachments and notes may reach this function, so we should handle them gracefully. I'm thinking that we should return `null` in those cases. I don't think we ever intended on citing file attachment, but I'm not sure about notes.